test_shopcart.py

```diff
@@ -28,6 +28,11 @@ def test_index(self):
         self.assertEqual( resp.status_code, status.HTTP_200_OK )
         self.assertTrue ('Shopcart Demo REST API Service' in resp.data)
 
+    def test_list_shopcarts(self):
+        resp = self.app.get('/shopcarts')
+        self.assertEqual( resp.status_code, status.HTTP_200_OK )
+        self.assertTrue( len(resp.data) > 0 )
```

Comment:
You are right. In general, when we do unit test, we don't run it against our real server. So we should know exactly what data are there to test, because we tell the test server to load the data before each test. We are not distinguish between the test server and real server at this moment, since no database is used and all objects lives in memory.